app/services/aspace_client.rb

```diff
@@ -0,0 +1,183 @@
+# frozen_string_literal: true
+
+require 'uri'
+require 'net/http'
+
+# Client for making requests via the ArchivesSpace API
+class AspaceClient
+  attr_reader :url
+
+  def initialize(url: ENV.fetch('ASPACE_URL', nil),
+                 user: ENV.fetch('ASPACE_USER', nil),
+                 password: ENV.fetch('ASPACE_PASSWORD', nil))
+    unless url && user && password
+      raise ArgumentError,
+            'Please provide the url, user, and password for ArchivesSpace'
+    end
+
+    @url = url
+    @user = user
+    @password = password
+  end
+
+  # Get a list of repositories
+  # See https://archivesspace.github.io/archivesspace/api/#get-a-list-of-repositories
+  def repositories
+    response = authenticated_get('repositories')
+
+    JSON.parse(response)
+  end
+
+  # Get the EAD XML representation of a resource
+  # See https://archivesspace.github.io/archivesspace/api/#get-an-ead-representation-of-a-resource
+  # @example client.resource_description('/repositories/2/resources/5363')
+  # @param [String] resource_uri the URI of the resource in ArchivesSpace
+  def resource_description(resource_uri)
```

Comment:
This is super minor -- but the only thing I find slightly inconsistent in this class is that for this method, we expect the constructed url to be passed as an argument. Whereas further down in `published_resource_uris`, we accept `repository_id` as its own argument -- also in the `AspaceQuery` class we also take `repository_id` and  `resource_id` as distinct arguments and then construct the URL internally in the methods.

Reply:
There's a method to my madness: the `resource_description` method will be called with data from the API response which is already in the form we need `/repositories/2/resources/5363`. In the other cases we're starting with more limited information. 🤷 If it bugs us enough we can refactor.